app/lib/nhs/immunisations_api.rb

```diff
@@ -239,9 +239,12 @@ def delete_immunisation(vaccination_record)
         # experiments show (by deleting and then re-creating a vaccination
         # record with an "update") that it appears that the e-tag is incremented
         # on the reviving update.
-        vaccination_record.update_columns(
+        vaccination_record.update!(
+          nhs_immunisations_api_id: nil,
+          nhs_immunisations_api_primary_source: nil,
           nhs_immunisations_api_synced_at: Time.current
         )
+        vaccination_record.save!(touch: false)
```

Comment:
Do we need to call this line if `update!` is being called above? Also worth noting that `update!` will set update the `updated_at` value, whereas it looks like we weren’t doing this before with `update_columns`.